### PR TITLE
OOP school library: unit tests

### PR DESCRIPTION
**Pull request by tumainimaganiko**
In this project, we did the following:

- Write unit tests using Rspec.
- Add unit tests for all the methods in all the classes.

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      expect(person.name).to eq('John Doe')
+    end
+  end
+
+  describe '#correct_name' do
+    it 'returns the correct name of the person' do
+      expect(person.correct_name).to eq('John Doe')
+    end
+  end
+end
```
Comment (AmaduKamara) [inline spec/person_spec.rb]: - [ ] Good job so far in writing a few tests for this file, but you are missing out on some other methods present in the person.rb file that you're required to write tests for. According to the requirement, you're expected to write tests for all methods present in all the classes. Kindly add some more tests for the add_rental, can_use_services, and of_age methods that are in the person.rb file to correspond things as expected 👍🏽

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✅ You used Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽
✅ Write tests in a separate directory for all files as required 👍🏽

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: spec/person_spec.rb (hunk 1 above)
Comment (Reggeditt) [reply spec/person_spec.rb]: fixed

File: spec/person_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      expect(person.can_use_services?).to eq(true)
+    end
+  end
+
+  describe '#correct_name' do
+    it 'returns the correct name of the person' do
+      expect(person.correct_name).to eq('John Doe')
+    end
+  end
+end
```
Comment (Munsa1) [inline spec/person_spec.rb]: - [ ] As requested by the previous reviewer, kindly test all methods in the `person.rb` file. Here's a way to directly test the `of_age?` method using the RSpec framework and reflection:


```
describe '#of_age?' do
    it 'returns true if the person is of age' do
      person = Person.new(20, 'John Doe', parent_permission: true)
      result = person.send(:of_age?) # Using reflection to access private method
      expect(result).to eq(true)
    end

    it 'returns false if the person is underage' do
      person = Person.new(16, 'Jane Doe', parent_permission: false)
      result = person.send(:of_age?) # Using reflection to access private method
      expect(result).to eq(false)
    end
  end
```

### Note that this test works, kindly inmplement it
image

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @tumainimaganiko @Reggeditt , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
Passing Tests :heavy_check_mark: 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DeliceLydia — APPROVED** [review]
## Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 

## Highlights
- Addressed changes requested



### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**